### PR TITLE
Add innovation flow uuid to create challenge/opportunity

### PR DESCRIPTION
**Pull request by alstojanovic**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/collaboration/opportunity/dto/opportunity.dto.create.ts [hunk 1]
```diff
@@ -6,4 +6,7 @@ import { UUID } from '@domain/common/scalars';
 export class CreateOpportunityInput extends CreateBaseChallengeInput {
   @Field(() => UUID, { nullable: false })
   challengeID!: string;
+
+  @Field(() => UUID, { nullable: false })
+  lifecycleID!: string;
```
Comment (valentinyanakiev) [inline src/domain/collaboration/opportunity/dto/opportunity.dto.create.ts]: Should this be mandatory?

File: src/domain/challenge/challenge/dto/challenge.dto.create.ts [hunk 2]
```diff
@@ -11,4 +11,7 @@ export class CreateChallengeInput extends CreateBaseChallengeInput {
   })
   @IsOptional()
   leadOrganizations?: string[];
+
+  @Field(() => UUID, { nullable: false })
+  lifecycleID!: string;
```
Comment (valentinyanakiev) [inline src/domain/challenge/challenge/dto/challenge.dto.create.ts]: Should this be mandatory?

File: src/domain/template/lifecycle-template/lifecycle.template.interface.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ export abstract class ILifecycleTemplate extends ITemplateBase {
   type!: string;
 
   @Field(() => LifecycleDefinitionScalar, {
-    nullable: true,
+    nullable: false,
```
Comment (valentinyanakiev) [inline src/domain/template/lifecycle-template/lifecycle.template.interface.ts:15]: Should this be mandatory?

File: src/domain/template/lifecycle-template/lifecycle.template.service.ts [hunk 4]
```diff
@@ -83,4 +84,44 @@ export class LifecycleTemplateService {
...
+  public async isLifecycleTemplateInHub(
+    lifecycleTemplateID: string,
+    hubID: string,
+    templateType: string
+  ) {
+    const [queryResult]: {
+      hubId: string;
+    }[] = await getConnection().query(
+      `
+      SELECT \'hub\'.\'id\' as \'hubId\' FROM \'hub\'
```
Comment (valentinyanakiev) [inline src/domain/template/lifecycle-template/lifecycle.template.service.ts]: You can have the result of the query (e.g. SELECT COUNT + WHERE hubId =...) already showing you whether isLifecycleTemplateInHub.

File: src/domain/template/lifecycle-template/lifecycle.template.interface.ts (hunk 3 above)
Comment (alstojanovic) [reply src/domain/template/lifecycle-template/lifecycle.template.interface.ts:15]: A template without a definition is nothing, the definition is the main information of the template. When we want a lifecycle template we want its definition.

File: src/domain/challenge/challenge/dto/challenge.dto.create.ts (hunk 2 above)
Comment (alstojanovic) [reply src/domain/challenge/challenge/dto/challenge.dto.create.ts]: Not sure. I asked Neil. It looks as it can be optional.

Comment (hero101) [reply src/domain/challenge/challenge/dto/challenge.dto.create.ts]: This must be `lifecycleTemplateId`

**Comment by Comoque1** [comment]
Add validation for scenario, when there are only 2 templates, and Admin updates the `type` of templateOne to be the same as the `type` of templateTwo.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**